### ISSUE TITLE
Building TensorFlow 1.2 from source results in catastrophic error?

### ISSUE THREAD

**Issue by kwotsin** [tensorflow/tensorflow#10434]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04 LTS
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.2
- **Bazel version (if compiling from source)**: 4.5.1
- **CUDA/cuDNN version**: 8.0/5.1
- **GPU model and memory**: GTX 860M
- **Exact command to reproduce**: `bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package --verbose_failures`

### Describe the problem
I have successfully installed TensorFlow 1.2 by upgrading the 1.1 version via Pip, but I wanted to install 1.2 from source to get the full benefits. However, strangely I received this error when building the pip package from bazel.

INFO: From Compiling tensorflow/core/kernels/batch_norm_op_gpu.cu.cc:
external/eigen_archive/unsupported/Eigen/CXX11/src/Tensor/TensorBroadcasting.h(271): internal error: assertion failed at: "/dvs/p4/build/sw/rel/gpu_drv/r361/r361_00/drivers/compiler/edg/EDG_4.10/src/folding.c", line 9819

```
1 catastrophic error detected in the compilation of "/tmp/tmpxft_00007cc2_00000000-7_batch_norm_op_gpu.cu.cpp1.ii".
Compilation aborted.
Aborted (core dumped)
ERROR: /home/kwotsin/tensorflow_official/tensorflow/tensorflow/core/kernels/BUILD:2730:1: output 'tensorflow/core/kernels/_objs/batch_norm_op_gpu/tensorflow/core/kernels/batch_norm_op_gpu.cu.pic.o' was not created.
ERROR: /home/kwotsin/tensorflow_official/tensorflow/tensorflow/core/kernels/BUILD:2730:1: not all outputs were created or valid.
Target //tensorflow/tools/pip_package:build_pip_package failed to build

```

I've not seen this error before - is it because I did not git check out the 1.2 version but instead the master branch?

**Comment by martinwicke** [comment]
If you didn't check out 1.2 you will get master, not 1.2, which contains strictly more stuff.

What compiler are you using to build this? Because this compilation error happens while compiling a CUDA file, maybe there's a problem with the nvcc version you're using (assuming you're using nvcc).

**Comment by sunsided** [comment]
I get the same error when building with CUDA 8.0 and cuDNN 5.1.5 on Nvidia Jetson TX1 (ARM) from the `r1.0` branch.

**Comment by martinwicke** [comment]
XQ, can you look at this? It looks like an internal compiler error in nvcc, but we must have triggered it only recently.

**Comment by zheng-xq** [comment]
Adding @benbarsdell, does this look like a NVCC problem?

**Comment by benbarsdell** [comment]
Yes, looking into this now.
@kwotsin could you provide the TF commit hash that you're on as well as the output of "nvcc --version"?

**Comment by jeichel-miovision** [comment]
I'm also seeing the same issue when building with CUDA 8.0, cuDNN 5.1.5 on Nvidia Jetson TX1 (ARM)

git commit (v1.1.0)
1ec6ed51182adf8f1b03a3188c16cd8a45ca6c85

nvcc --version
```
Built on Fri_Jul_15_14:52:12_CDT_2016
Cuda compilation tools, release 8.0, V8.0.33
```

**Comment by jeichel-miovision** [comment]
note for TX1 users (https://devtalk.nvidia.com/default/topic/987306/jetson-tx1/internal-error-in-nvidia-driver-code-while-building-r0-12-or-master-tensorflow-on-tegra-tx1/2)

At least for Tensorflow r0.12, NVIDIA released a CUDA 8.0.64 version with Jetpack 3.0 for the TX2. From their forum it looked like they had issues with publicly released versions of CUDA 8.0.34 and nvcc for the TX1 that were NOT yet resolved.

@kwotsin not sure if this applies to you, but is the full version of CUDA that you are using?
@sunsided you're probably in the same boat as me on the TX1. I would try to contact NVIDIA and see if they have a fix that you can use. I have access to a TX2 that I will test with next.

**Comment by kwotsin** [comment]
@jeichel-miovision What is the full version of CUDA you're referring to? I installed my CUDA quite some time ago so I suspect it's an earlier version of CUDA 8.0. FYI my error is from building on a laptop, not the TX1. 

@martinwicke I have tried to checkout with r1.2, and used `bazel clean` before I configured and built with bazel. 

I now have an error that goes like this:

```
external/eigen_archive/unsupported/Eigen/CXX11/src/Tensor/TensorBroadcasting.h(275): internal error: assertion failed at: "/dvs/p4/build/sw/rel/gpu_drv/r361/r361_00/drivers/compiler/edg/EDG_4.10/src/folding.c", line 9819


1 catastrophic error detected in the compilation of "/tmp/tmpxft_00001223_00000000-7_softmax_op_gpu.cu.cpp1.ii".
Compilation aborted.
Aborted (core dumped)
```
@benbarsdell Here is the output of my `nvcc --version`

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2016 NVIDIA Corporation
Built on Fri_Jul_15_14:46:45_CDT_2016
Cuda compilation tools, release 8.0, V8.0.33

```
Also, how do you check the TF commit hash?

**Comment by jeichel-miovision** [comment]
@kwotsin your nvcc --version actually posted the full release code: V8.0.33 
If possible, I would strongly recommend upgrading to the latest cuda version, >= V8.0.64, since there are known issues with the older versions.

for those interested in the jetson progress:
I've also successful built TF v1.1.0 (1ec6ed51182adf8f1b03a3188c16cd8a45ca6c85) on the TX2 using CUDA V8.0.64 with bazel 0.5.1. 

I followed the EXACT same process as I did on my TX1, and as far as I can tell, I suspect the only major difference points to the difference in CUDA version.

Specifically for the TX1 / TX2 platform, I still had to patch TF v1.1.0 by making the modifications in lines 65 to 75 here: https://github.com/jetsonhacks/installTensorFlowTX2/blob/master/patches/tensorflow.patch

I was then able to import tensorflow into python and run some basic test code to validate the installation.

**Comment by kwotsin** [comment]
@jeichel-miovision thanks for your advice. I upgraded my cuda version and now installing from source is fixed. I have tried installing a version of TF higher than 1.0 in TX1 before but didn't manage to install it successfully. Perhaps it was because I used the older cuda version that came with jetpack 2.3.1.

**Comment by sugartom** [comment]
@jeichel-miovision, @kwotsin  Hi, may I ask how you upgrade your cuda on TX1/TX2? I used JetPack to install CUDA v8.0.33, and has seen the same error message `external/eigen_archive/unsupported/Eigen/CXX11/src/Tensor/TensorBroadcasting.h(275): internal error: assertion failed at: "/dvs/p4/build/sw/rel/gpu_drv/r361/r361_00/drivers/compiler/edg/EDG_4.10/src/folding.c", line 9819`

Thanks in advance! :-)

**Comment by jeichel-miovision** [comment]
@sugartom I was not able to get the TX1 beyond 8.0.33 (but I didn't look into this aspect very hard). 

Instead I downloaded  JetPack 3.0 from NVIDIA (https://developer.nvidia.com/embedded/jetpack). When I installed CUDA for my TX2, the installer downloaded "cuda-repo-l4t-8-0-local_8.0.64-1_arm64.deb" into my jetpack_download folder and proceeded to install version 8.0.64 onto my TX2.

I'm not sure how, or if it is currently possible, to install 8.0.64 on the TX1. 
From NVIDIA (https://developer.nvidia.com/embedded/jetpack)
> JetPack 3.0 adds support for the new Jetson TX2 Developer Kit with a preview release of L4T with kernel 4.4, and continues to support the Jetson TX1 Developer Kit and the Jetson TK1 Developer Kit. Components for Jetson TX1 and Jetson TK1 Developer Kits remain unchanged from the previous version of JetPack. (A future release of JetPack will support both Jetson TX2 and Jetson TX1 with aligned API and Linux kernel versions.)

**Comment by jeichel-miovision** [comment]
@sugartom I also running an experiment and manually copied the cuda-repo-l4t-8-0-local_8.0.64-1_arm64.deb package to my TX1. `sudo dpkg -i cuda-repo-l4t-8-0-local_8.0.64-1_arm64.deb && sudo apt-get update && sudo apt-get upgrade`

I'll try rebuilding tensorflow v1.1.0 throughout the day and post the results.

**Comment by kwotsin** [comment]
@jeichel-miovision @sugartom I wasn't able to upgrade the cuda in the past due to some architectural issues. Basically the TX1 says it's using aarch64 and so even with the arm64 deb I couldn't install it. Perhaps a successful upgrade of CUDA would make it possible to install tensorflow above v1.0, which would enable a lot of other operations like quantization. thanks to @jeichel-miovision in advance for trying out!

**Comment by eddytrex** [comment]
Hi, someone has try to compile it with the new Jetson Pack 3.1 ? 

I have this error.

```bash

INFO: Found 1 target...
INFO: From Compiling external/llvm/lib/MC/MachObjectWriter.cpp [for host]:
external/llvm/lib/MC/MachObjectWriter.cpp: In member function 'void llvm::MachObjectWriter::writeNlist(llvm::MachObjectWriter::MachSymbolData&, const llvm::MCAsmLayout&)':
external/llvm/lib/MC/MachObjectWriter.cpp:376:39: warning: 'AliaseeInfo' may be used uninitialized in this function [-Wmaybe-uninitialized]
     Address = AliaseeInfo->StringIndex;
                                       ^
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-invalid-partial-specialization'
INFO: From Compiling external/llvm/lib/MC/WasmObjectWriter.cpp [for host]:
external/llvm/lib/MC/WasmObjectWriter.cpp: In member function 'virtual void {anonymous}::WasmObjectWriter::writeObject(llvm::MCAssembler&, const llvm::MCAsmLayout&)':
external/llvm/lib/MC/WasmObjectWriter.cpp:802:27: warning: 'Index' may be used uninitialized in this function [-Wmaybe-uninitialized]
       Export.Index = Index;
                           ^
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-invalid-partial-specialization'
[2,311 / 6,048] Compiling external/llvm/utils/TableGen/CodeGenRegisters.cpp [for host]

Server terminated abruptly (error code: 14, error message: '', log file: '/home/nvidia/.cache/bazel/_bazel_nvidia/d2751a49dacf4cb14a513ec663770624/server/jvm.out')

```

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by martinwicke** [comment]
I believe this is fixed, but I cannot say for sure. Can you reopen if this still occurs with a more recent TensorFlow?

**Comment by allendred** [comment]
I get the same error when building with CUDA 8.0 and cuDNN 7.0 on ubuntu16.04 from the r1.13 branch.
device：1080ti bazel version ：0.20
